frontend/src/aws-exports.js

```diff
@@ -1,15 +1,15 @@
 const env = JSON.parse(window.localStorage.getItem('env_data'))
 
 const config = {
-    aws_cognito_region: env.fam_cognito_region.value,
-    aws_user_pools_id: env.fam_user_pool_id.value,
-    aws_user_pools_web_client_id: env.fam_console_web_client_id.value, // This is App Client Id
+    aws_cognito_region: env?.fam_cognito_region.value,
+    aws_user_pools_id: env?.fam_user_pool_id.value,
+    aws_user_pools_web_client_id: env?.fam_console_web_client_id.value, // This is App Client Id

```

Comment:
Not sure why this was made optional. Nothing will work if the env data is missing. Does it make for nicer error messages or something?

Reply:
You always have good instinct Basil!

The solution so far is the only solution that works for me. I tried moving around Amplify initialization but nothing works as env.json always not catching up with initialization (loading timing). I seemed once see Vue/Vite talk about Webpack preload setting but was hesitant to try (I think we don't use webpack with Vite or maybe internally it is, don't know).

To answer you question, yes, I faced problem when moving Amplify init into AuthService component and break the test and had to use this "?/optional" for the aws-exports.js configuration. Then when pipeline was running it failed (test pass) on timeout from Amplify and showing message that makes sense more (something like it can not initialize with Cognito). So leaving it optional there would make down stream error to show up nicely.

Reply:
Works for me then. I'd suggest adding a comment explaining this then, so future developers aren't left wondering.

Reply:
Comment added. Thanks Basil.